File: check combination C or CD or CDD.c
/*
INPUT:
CDCDDCCCCDD
OUTPUT:
yes
INPUT:
CCCCCDCDDD
OUTPUT:
NO
*/
#include<stdio.h>
#include <stdlib.h>

int main()
{int i;
 char s[101];
 scanf("%s",s);
 for(i=0;i<strlen(s);i++)
 {

     if(s[i] == 'C' && s[i+1] == 'D' && s[i+2] == 'D')
      {
          i+=2;
          continue;
      }
      else if(s[i] == 'C' && s[i+1] == 'D')
      {
          i++;
          continue;
      }
      else if(s[i] == 'C')
      {
          continue;
      }
      else
      {
          printf("no");
          return;
      }
 }
 printf("yes");
}

File: check combination C or CD or CDD.java
/*
INPUT:
CDCDDCCCCDD
OUTPUT:
yes
INPUT:
CCCCCDCDDD
OUTPUT:
NO
*/
import java.util.*;
public class Hello {

    public static void main(String[] args) {
		String s=new Scanner(System.in).next();
		s=s.replaceAll("CDD","").replaceAll("CD","").replaceAll("C","");
		System.out.print((s.length()==0)?"yes":"no");

	}
